generate/report.go

```diff
@@ -288,3 +311,25 @@ func (s dirReport) empty() bool {
 		len(s.deleted) == 0 &&
 		s.err == nil
 }
+
+type result interface {
+	Result | FailureResult
+}
+
+func joinResults[T result](results ...[]T) []T {
```

Comment:
Maybe just do [T any] and drop the constraint ? There is nothing on the algorithm that imposes a constraint on the type.